src/main/js/approved/ApprovedCertificateDetailsModal.tsx

```diff
@@ -0,0 +1,175 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React, { FC, useState } from "react";
+import { apiClient, Button, ErrorNotification, Level, Modal } from "@scm-manager/ui-components";
+import { useTranslation } from "react-i18next";
+import { Certificate, formatAsTimestamp, parseCommonNameFromDN } from "../certificates";
+import styled from "styled-components";
+import classNames from "classnames";
+import { Link } from "@scm-manager/ui-types";
+
+export const SizedModal = styled(Modal)`
+  .modal-card {
+    width: 95%;
+    height: auto;
+  }
+
+  th {
+    width: 20%;
+  }
+`;
+
+export const StyledTreeElement = styled.li<{ depth: number }>`
+  margin-left: ${props => props.depth}rem;
+  width: fit-content;
+`;
+
+export const ChainButton = styled(Button)`
+  background: none !important;
+  border: none !important;
+  box-shadow: none !important;
+  padding: 0;
+  text-decoration: none;
+  cursor: pointer;
+`;
+
+export type ChainEntryProps = {
+  certificate: Certificate;
+  depth: number;
+  selected: boolean;
+  onClick: () => void;
+};
+
+export const ChainEntry: FC<ChainEntryProps> = ({ certificate, depth, selected, onClick }) => (
+  <StyledTreeElement depth={depth * 3} onClick={onClick}>
+    <ChainButton icon="certificate" className={classNames("is-inverted", { "is-link": selected })}>
+      {parseCommonNameFromDN(certificate.subjectDN)}
+    </ChainButton>
+  </StyledTreeElement>
+);
+
+type Props = {
+  active: boolean;
+  onClose: () => void;
+  certificate: Certificate;
+  refresh: () => void;
+};
+
+const ApprovedCertificateDetailsModal: FC<Props> = ({ onClose, certificate, active, refresh }) => {
+  const [t] = useTranslation("plugins");
+  const chain = [certificate, ...certificate._embedded.chain].reverse();
+  const [selectedCert, setSelectedCert] = useState<Certificate>(certificate);
+  const [error, setError] = useState<Error>();
+
+  const manageCertificate = (link: string) => {
+    apiClient
+      .post(link)
+      .then(() => onClose())
+      .then(() => refresh())
+      .catch(setError);
+  };
```

Comment:
This function should be the return value of a custom hook and a loading state is missing.